File: NG-Training/NG.NET/TaskTrackerApp/src/app/interfaces/compliance-dto.ts
export interface ComplianceDTO {
    id: string; // GUIDs are typically represented as strings in TypeScript
    title: string;
    description: string;
    requiredPercentage: number; // Matches the C# double
    createdBy: string;
    createdDate: string; // DateTimes are often represented as ISO strings
    questionsCount: number;
}

File: .NET-Training/Backend-Training/StandardApiSolution/KnowledgeBase/Models/DTOs/ComplianceDTO.cs
namespace KnowledgeBaseApi.Models.DTOs
{
    public class ComplianceDTO
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public string Title { get; set; }
        public string Description { get; set; }
        public double RequiredPercentage { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedDate { get; set; }
        public int QuestionsCount { get; set; }
    }
}
